index.js

```diff
@@ -35,8 +116,18 @@ Changelog.prototype.parse = function( callback ) {
 	}.bind( this ));
 };
 
-Changelog.prototype.ticketUrl = function( id ) {
-	return this.options.ticketUrl.replace( "{id}", id );
+Changelog.prototype.ticketUrl = function( id, ticketType ) {
+	if ( !ticketType ) {
+		ticketType = this.ticketTypes[0];
+	}
+
+	return this.ticketUrlTemplates[ ticketType ].replace( "{id}", id );
+};
+
+Changelog.ticketUrlFactory = function( ticketType, thisArg ) {
```

Comment:
I don't know why you need a wrapping function in the first place.You're generating a function that has one line of code in it.